File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        // map<int, int> mp;

        // for(auto val : nums){
        //     mp[val]++;
        // }

        // for(auto val : mp){
        //     if(val.second > 1){
        //         return true;
        //     }
        // }

        // return false;

        set<int> st(nums.begin() , nums.end());
        return st.size() != nums.size();
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> hashmap = new HashSet<>();

        for(int val : nums){
            if(hashmap.contains(val)){
                return true;
            }

            hashmap.add(val);
        }

    return false;
    }
}